secure/key-manager/src/counters.rs

```diff
@@ -13,6 +14,40 @@ static STATE_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| {
     .unwrap()
 });
 
-pub fn increment_state(key: &str, state: &str) {
-    STATE_COUNTER.with_label_values(&[key, state]).inc();
+/// Metric counter keys.
+const CHECK_KEYS: &str = "check_keys";
+const CONSENSUS_KEY: &str = "consensus_key";
+
+/// Metric counter states.
+pub const KEYS_STILL_FRESH: &[&str] = &[CHECK_KEYS, "keys_still_fresh"];
+pub const LIVENESS_ERROR_ENCOUNTERED: &[&str] = &[CHECK_KEYS, "liveness_error_encountered"];
+pub const NO_ACTION: &[&str] = &[CHECK_KEYS, "no_action"];
+pub const ROTATED_IN_STORAGE: &[&str] = &[CONSENSUS_KEY, "rotated_in_storage"];
+pub const SUBMITTED_ROTATION_TRANSACTION: &[&str] =
+    &[CONSENSUS_KEY, "submitted_rotation_transaction"];
+pub const WAITING_ON_RECONFIGURATION: &[&str] = &[CHECK_KEYS, "waiting_on_reconfiguration"];
+pub const WAITING_ON_TRANSACTION_EXECUTION: &[&str] =
+    &[CHECK_KEYS, "waiting_on_transaction_execution"];
+pub const UNEXPECTED_ERROR_ENCOUNTERED: &[&str] = &[CHECK_KEYS, "unexpected_error_encountered"];
+
+/// Initializes all metric counter states.
+pub fn initialize_all_metric_counters() {
+    let metric_counter_states = &[
+        KEYS_STILL_FRESH,
+        LIVENESS_ERROR_ENCOUNTERED,
+        NO_ACTION,
+        ROTATED_IN_STORAGE,
+        SUBMITTED_ROTATION_TRANSACTION,
+        WAITING_ON_RECONFIGURATION,
+        WAITING_ON_TRANSACTION_EXECUTION,
+        UNEXPECTED_ERROR_ENCOUNTERED,
+    ];
+    let _ = metric_counter_states.iter().map(|metric_counter_state| {
+        COUNTER.with_label_values(metric_counter_state).reset();
+    });
+}
```

Comment:
Oh! Sorry I missed this. This won't actually initialize anything because this is just creating a lazy iterator. You should be able to change `map` to `for_each` or use a normal for loop to get the desired behavior.

/cancel

Reply:
Ooh, thanks @bmwill, used "for_each". Kinda annoying that I can't actually test the functionality until it lands because spawning up a k8s environment from scratch is a real pain.. 😄